src/shared/components/plots/BoxScatterPlot.tsx

```diff
@@ -1014,3 +1014,72 @@ export function toBoxPlotData<D extends IBaseBoxScatterPlotPoint>(
             );
         });
 }
+
+export function toDataDescriptive<D extends IBaseBoxScatterPlotPoint>(
+    data: IBoxScatterPlotData<D>[],
+    logScale?: IAxisLogScaleParams
+) {
+    return data.map(d => {
+        const scatterValues = d.data.map(x =>
+            logScale ? logScale.fLogScale(x.value, 0) : x.value
+        );
+        const count = scatterValues.length;
+        // Calculate minimum and maximum
+        const minimum = Math.min(...scatterValues);
+        const maximum = Math.max(...scatterValues);
+        // Calculate mean
+        const total = scatterValues.reduce(
+            (sum: number, value: number) => sum + value,
+            0
+        );
```

Comment:
After we changed the mean calculation to `_.mean`, I think we can safely remove these lines